agent/command/git.go

```diff
@@ -529,6 +582,10 @@ func (c *gitFetchProject) fetchSource(ctx context.Context,
 	attempt := 0
 	return c.retryFetch(ctx, logger, true, opts, func(opts cloneOpts) error {
 		attempt++
+		if attempt > 1 {
```

Comment:
We really start at attempt = 1 since we increment in the beginning of the function. I.e. the first run of the function will have attempt = 1. Then on the first retry, attempt = 2 (causing it to fallback).